connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -687,101 +617,72 @@ public void revokeAndReassign(boolean incompleteRebalance) throws TimeoutExcepti
             );
             expectConfigRefreshAndSnapshot(secondSnapshot);
         }
-        member.requestRejoin();
-        PowerMock.expectLastCall();
+
+        doNothing().when(member).requestRejoin();
+
+        herder.tick();
 
         // re-assign the connector back to the same worker to ensure state was cleaned up
         expectRebalance(configOffset, Arrays.asList(CONN1), Arrays.asList());
-        Capture<Callback<TargetState>> onSecondStart = newCapture();
-        worker.startConnector(EasyMock.eq(CONN1), EasyMock.anyObject(),
-            EasyMock.anyObject(),
-            EasyMock.eq(herder), EasyMock.eq(TargetState.STARTED), capture(onSecondStart));
-        PowerMock.expectLastCall().andAnswer(() -> {
-            onSecondStart.getValue().onCompletion(null, TargetState.STARTED);
-            return true;
-        });
-        member.wakeup();
-        PowerMock.expectLastCall();
-        expectExecuteTaskReconfiguration(true, conn1SinkConfig, () -> TASK_CONFIGS);
-        member.poll(EasyMock.anyInt());
-        PowerMock.expectLastCall();
-
-        PowerMock.replayAll();
 
         herder.tick();
-        if (incompleteRebalance) {
-            herder.tick();
-        }
-        herder.tick();
-        herder.tick();
 
-        PowerMock.verifyAll();
+        verify(worker, times(2)).startConnector(eq(CONN1), any(), any(), eq(herder), eq(TargetState.STARTED), any());
+        verify(worker, times(2)).connectorTaskConfigs(eq(CONN1), eq(conn1SinkConfig));
+        verify(worker).startSourceTask(eq(TASK1), any(), any(), any(), eq(herder), eq(TargetState.STARTED));
+
```

Comment:
This test case was introduced in https://github.com/apache/kafka/pull/9765, which addressed a bug in incremental rebalancing that caused duplicate connectors/tasks on a worker, caused by failure to stop the connectors/tasks during a rebalance.

It currently passes without the fix from https://github.com/apache/kafka/pull/9765, but in the same situation, fails if we add an assertion about stopping the connector:

```suggestion
        verify(worker).stopAndAwaitConnector(CONN1);

```

Reply:
Wow, good catch, and thanks for the additional context!